x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/components/processor_form/processors/shared.ts

```diff
@@ -69,6 +81,41 @@ export const from = {
   optionalArrayOfStrings: (v: string[]) => (v.length ? v : undefined),
   undefinedIfValue: (value: unknown) => (v: boolean) => (v === value ? undefined : v),
   emptyStringToUndefined: (v: unknown) => (v === '' ? undefined : v),
+  /**
+   * Useful when serializing user input from a <textarea /> that we want to later JSON.stringify but keep the same as what
+   * the user input. For instance, given "my\ttab", encoded as "my%5Ctab" will JSON.stringify to "my\\ttab", instead we want
+   * to keep the input exactly as the user entered it.
+   */
+  unescapeBackslashes: (v: unknown) => {
+    if (typeof v === 'string') {
+      try {
+        return JSON.parse(`"${v}"`);
+      } catch (e) {
+        // Best effort
+        return v;
+      }
+    }
+  },
+};
+
+const isJSONString = (v: string) => {
+  try {
+    JSON.parse(`"${v}"`);
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+
+export const isJSONStringValidator: ValidationFunc = ({ value }) => {
```

Comment:
Could you leverage https://github.com/elastic/kibana/blob/master/src/plugins/es_ui_shared/static/validators/string/is_json.ts here?

Reply:
Yeah I looked at that validator, but the condition for returning `true` also requires that the resulting of `JSON.parse` be an object - in this case it will be a `string`.

https://github.com/elastic/kibana/blob/52a1ce17233079138ee4cf6bd9f86eaa6266ea7a/src/plugins/es_ui_shared/static/validators/string/is_json.ts#L12

For now I think it is OK to keep that behaviour and create a our own validator here.